engines/dradis-api/app/controllers/dradis/ce/api/api_controller.rb

```diff
@@ -20,6 +22,12 @@ class APIController < ApplicationController
     # No CSRF protection for the wicked!
     protect_from_forgery with: :null_session
 
+    # Set 'whodunnit' in paper trail versions to be the email address of the
+    # current user
+    def user_for_paper_trail
+      current_user.email if current_user
```

Comment:
Do we need the `if current_user` ?

Shouldn't authentication be guaranteeing that a current_user is present before accepting changes to records?

Reply:
You are correct. Warden tries to authenticate the user and it won't reach here when no user is found. Removed the user check.